### PR TITLE
Introductory Cloud Shell Tutorial for Terraform Cloud Ops Agent Policy

### PR DESCRIPTION
**Pull request by kyleabenson**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./README.md:13:Use the interactive cloudshell tutorial to learn the basics.  
./tutorial/main.tf:41:    // This non production example uses the default compute service account. 
./tutorial/tutorial.md:3:This tutorial will guide you through using Terraform by Hashicorp to deploy three new VMs 
./tutorial/tutorial.md:5:## Prerequisites 
./tutorial/tutorial.md:8:Enter the tutorial directory. 
./tutorial/tutorial.md:9:```bash 
./tutorial/tutorial.md:13:Rename the Terraform Variable Example file.  
./tutorial/tutorial.md:25:If needed, login to your gcp account within cloudshell, this ensures you're able to run the necessary commands.  
./tutorial/tutorial.md:30:Next, set a project you'll be deploying VMs in. To see available projects:   
./tutorial/tutorial.md:34:And then set your configuration for whichever project you prefer:  
./tutorial/tutorial.md:43:```bash 
./tutorial/tutorial.md:47:## Service Account Setup 
./tutorial/tutorial.md:48:Next you'll need to create a service account and credentials file. These resources allow Terraform to create the VM. 
./tutorial/tutorial.md:49:**NOTE** 
./tutorial/tutorial.md:57:```bash 
./tutorial/tutorial.md:58:gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" --role="roles/compute.admin" 
./tutorial/tutorial.md:60:```bash 
./tutorial/tutorial.md:61:gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" --role="roles/osconfig.guestPolicyAdmin" 
./tutorial/tutorial.md:90:You'll make changes in the `terraform.tfvars` file. At a minimum you need to specify values for the following inside of double quotes (""):  
./tutorial/tutorial.md:93:service_account_email = 
./tutorial/tutorial.md:100:If you're unsure of your project number, you can capture it with the following command in CloudShell:  
./tutorial/terraform.tfvars.example:3:service_account_email = 
./tutorial/terraform.tfvars.example:5://gcp_region = 
./tutorial/terraform.tfvars.example:6://gcp_zone = 
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
Error: No newline at end of file ./tutorial/variables.tf
Error: No newline at end of file ./tutorial/tutorial.md
Error: No newline at end of file ./tutorial/terraform.tfvars.example
 ``` 
- ⚠️check_headers
 All files need a license header. Please make sure all your files include the appropriate header. A helper tool available [here](https://github.com/google/addlicense). 
 ```bash 
Checking file headers
2 files have incorrect boilerplate headers:
tutorial/main.tf
tutorial/variables.tf
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
tutorial/main.tf
--- old/tutorial/main.tf
+++ new/tutorial/main.tf
@@ -8,46 +8,46 @@
 
 // Terraform plugin for creating random ids
 resource "random_id" "instance_id" {
- byte_length = 8
+  byte_length = 8
 }
 
 
 // A Single Compute Engine instance
 resource "google_compute_instance" "default" {
- name         = "terraform-test-vm"
- machine_type = "f1-micro"
- zone         = var.gcp_zone
-
- boot_disk {
-   initialize_params {
-     image = "centos-cloud/centos-8-v20210817"
-   }
- }
- labels = {
-    env = "prod"
-    app = "myproduct"
+  name         = "terraform-test-vm"
+  machine_type = "f1-micro"
+  zone         = var.gcp_zone
+
+  boot_disk {
+    initialize_params {
+      image = "centos-cloud/centos-8-v20210817"
+    }
+  }
+  labels = {
+    env        = "prod"
+    app        = "myproduct"
     created_by = "terraform"
   }
 
- network_interface {
-   network = "default"
+  network_interface {
+    network = "default"
 
-   access_config {
-     // Include this section to give the VM an external ip address
-   }
- }
- service_account {
+    access_config {
+      // Include this section to give the VM an external ip address
+    }
+  }
+  service_account {
     // Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
     // This non production example uses the default compute service account. 
     email  = var.service_account_email
     scopes = ["cloud-platform"]
- }
+  }
 }
 
 // Cloud Ops Agent Policy
 module "agent_policy" {
-  source     = "terraform-google-modules/cloud-operations/google//modules/agent-policy"
-  version    = "~> 0.1.0"
+  source  = "terraform-google-modules/cloud-operations/google//modules/agent-policy"
+  version = "~> 0.1.0"
 
   project_id = var.gcp_project
   policy_id  = "ops-agents-example-policy"
@@ -61,8 +61,8 @@
   ]
   group_labels = [
     {
-      env = "prod"
-      app = "myproduct"
+      env        = "prod"
+      app        = "myproduct"
       created_by = "terraform"
     }
   ]
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
tutorial/variables.tf
--- old/tutorial/variables.tf
+++ new/tutorial/variables.tf
@@ -5,17 +5,17 @@
 }
 
 variable "gcp_zone" {
-    description = "GCP Zone"
-    type        = string
-    default     = "us-central1-a"
+  description = "GCP Zone"
+  type        = string
+  default     = "us-central1-a"
 }
 
 variable "gcp_project" {
-    description = "GCP Project"
-    type        = string
+  description = "GCP Project"
+  type        = string
 }
 
 variable "service_account_email" {
-    description = "Service Account Email"
-    type        = string
+  description = "Service Account Email"
+  type        = string
 }
\ No newline at end of file
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

File: tutorial/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+```
+
+## Service Account Setup 
+Next you'll need to create a service account and credentials file. These resources allow Terraform to create the VM. 
+**NOTE** 
+For this example we'll name the service account and display it as 'terraform' but this is an arbitrary string and can be subsituted however you see fit.
+
+1. Create the Terraform service account.
+```bash
+gcloud iam service-accounts create terraform --description="Service account for VM provisioning with Terraform" --display-name="terraform"
```
Comment (qingling128) [inline tutorial/tutorial.md:54]:
```suggestion
gcloud iam service-accounts create terraform \
  --description="Service account for VM provisioning with Terraform" \
  --display-name="terraform"
```

File: tutorial/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+```
+2. Create the iam policy bindings and attach the necessary role to the account:
+```bash 
+gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" --role="roles/compute.admin"
```
Comment (qingling128) [inline tutorial/tutorial.md:58]:
```suggestion
gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT \
  --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" \
  --role="roles/compute.admin" 
```

File: tutorial/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+1. Create the Terraform service account.
+```bash
+gcloud iam service-accounts create terraform --description="Service account for VM provisioning with Terraform" --display-name="terraform"
+```
+2. Create the iam policy bindings and attach the necessary role to the account:
+```bash 
+gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" --role="roles/compute.admin" 
+```
+```bash 
+gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" --role="roles/osconfig.guestPolicyAdmin"
```
Comment (qingling128) [inline tutorial/tutorial.md:61]:
```suggestion
gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT \
  --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" \
  --role="roles/osconfig.guestPolicyAdmin" 
```

File: tutorial/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+```
+2. Create the iam policy bindings and attach the necessary role to the account:
+```bash 
+gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" --role="roles/compute.admin" 
+```
+```bash 
+gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" --role="roles/osconfig.guestPolicyAdmin" 
+```
+```bash
+gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" --role="iam.serviceAccountUser"
```
Comment (qingling128) [inline tutorial/tutorial.md:64]:
```suggestion
gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT \
  --member="serviceAccount:terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com" \
  --role="iam.serviceAccountUser"
```

File: tutorial/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+```
+
+3. Create the key-file associated with the service account:
+```bash
+gcloud iam service-accounts keys create test-key.json --iam-account=terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com
```
Comment (qingling128) [inline tutorial/tutorial.md:69]:
```suggestion
gcloud iam service-accounts keys create test-key.json \
  --iam-account=terraform@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com
```

File: tutorial/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+gcloud auth login
+```
+
+Next, set a project you'll be deploying VMs in. To see available projects:   
+```bash
+gcloud projects list
+```
+And then set your configuration for whichever project you prefer:  
+```bash
+gcloud config set project PROJECT_ID
```
Comment (qingling128) [inline tutorial/tutorial.md:36]:
```suggestion

GOOGLE_CLOUD_PROJECT="{FILL IN THE PROJECT ID HERE}"
gcloud config set project $GOOGLE_CLOUD_PROJECT
```

**Pull request closed without merging**